docs/vector_search.rst

```diff
@@ -0,0 +1,278 @@
+
+    [1.0, 3.2, 1.3]
+
+The algorithm will then determine the most relevant photos by traversing through a graph and applying similarity algorithms such as cosine similarity, dot product, or others.
+
+You can find more detailed information about vector embeddings in the following articles:
+
+- `Google <https://cloud.google.com/blog/topics/developers-practitioners/meet-ais-multitool-vector-embeddings>`_
+- `Elastic <https://www.elastic.co/what-is/vector-embedding>`_
+
+Vector Search Applications
+--------------------------
+
+The following is a list of example applications that can benefit from vector search:
+
+- Photo search
+- Review search
+- Similar businesses search
+- Bot detection
+- Chatbots
+
+Vector Embeddings on Nrtsearch
+------------------------------
+
+Nrtsearch is based on the popular open-source search library, Lucene. Vector search support was added to Lucene in version 9+. It uses the Hierarchical Navigable Small Worlds (HNSW) graph to traverse vector data to find the most relevant results. It supports vector-only search as well as hybrid (vector + text search).
+
+More information about Lucene's support of vector search can be found at:
+
+- `Lucene Vector Search <https://www.apachecon.com/acna2022/slides/04_lucene_vector_search_sokolov.pdf>`_
+
+In the following sections, we will go through different steps to launch and configure an Nrtsearch cluster with vector search support.
+
+What was Already Supported?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Nrtsearch v0 already supports regular vector fields. However, it's not using any kind of graph data structure under the hood to do advanced KNN lookups in the recall/boosting phase. This data can only be used in the scoring phase to find similarity scores using custom functions/painless scripts.
+
+Launching Cluster with Vector Search Support
+============================================
+
+Estimate Cluster Size
+---------------------
+
+Before launching a cluster, you need to figure out how big the data will be.
+
+The general guidelines to calculate the size introduced by embeddings is to use the following formula::
+
+    Size in bytes = Total Num docs * Float Size * (Embeddings dimensions + hnsw additional storage)
+
+For example, if your index will have around 390M docs with embeddings of 512 dimensions, the total size for this cluster will be around 817GB::
+
+    => ~390M * 4 * (512 + 12) = ~817GB
+
+Note that the above formula is only for vector fields. Other fields in the index will require additional space.
+
+Configuring Cluster
+-------------------
+
+To create a cluster that supports vector search, you can follow the guide to launch regular Nrtsearch clusters with a few minor changes that are discussed below.
+
+Add the following vector search-specific settings to the index configuration under the datastore schema repo::
```

Comment:
This refers to internal documentation and repos

Reply:
Removed this